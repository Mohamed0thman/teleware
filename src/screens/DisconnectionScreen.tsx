import {StyleSheet, Text, View, Image} from 'react-native';
import React, {useState} from 'react';
import RootScreen from '../component/RootScreen';
import {FONTS, IMAGES, SCALE} from '../constants';
import {CustomButton, Typography} from '../component/UI';
import {useAppSelector} from '../store/configureStore';
import NetInfo from '@react-native-community/netinfo';
import {StackScreenProps} from '@react-navigation/stack';
import {RootStackParamList} from '../navigation/RootNavigation';
import {useTranslation} from 'react-i18next';

const {rh, rw} = SCALE;

type Props = StackScreenProps<RootStackParamList, 'Disconnection'> & {};

const DisconnectionScreen = ({navigation}: Props) => {
  const {isRtl} = useAppSelector(state => state.setting);
  const [isRetrying, setIsRetring] = useState<boolean>(false);

  const {t} = useTranslation();

  const onRetry = async () => {
    setIsRetring(true);
    const netInfo = await NetInfo.fetch();
    if (netInfo.isConnected) {
      navigation.replace('Home');
    }
    setIsRetring(false);
  };

  return (
    <RootScreen containerStyle={styles.rootScreen}>
      <Image
        source={IMAGES.disconctionImage}
        style={styles.image}
        resizeMode="contain"
      />
      <Typography isRtl={isRtl} style={{marginTop: rh(5), ...FONTS.h3}}>
        {t('app:titles.connectionError')}
      </Typography>
      <Typography style={{marginTop: rh(5), ...FONTS.h4}} isRtl={isRtl}>
        {t('app:titles.oops')}{' '}
      </Typography>
      <CustomButton
        label={t('app:buttons.retry')}
        onPress={onRetry}
        disabled={isRetrying}
        containerStyle={{marginTop: rh(20)}}
      />
    </RootScreen>
  );
};

export default DisconnectionScreen;

const styles = StyleSheet.create({
  rootScreen: {justifyContent: 'center', alignItems: 'center'},
  image: {width: '80%', height: '50%'},
});
