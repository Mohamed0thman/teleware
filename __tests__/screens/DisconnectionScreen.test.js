import React from 'react';

import DisconnectionScreen from '../../src/screens/DisconnectionScreen';

describe('languge screen', () => {
  it('renders correctly', () => {
    renderWithRedux(<DisconnectionScreen />);
  });

  it('renders correctly', () => {
    expect(renderWithRedux(<DisconnectionScreen />)).toMatchSnapshot();
  });
});
