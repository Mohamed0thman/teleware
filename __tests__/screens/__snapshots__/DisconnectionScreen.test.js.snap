// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languge screen renders correctly 1`] = `
<View
  style={
    [
      {
        "paddingBottom": 0,
        "paddingTop": 0,
      },
      {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../src/assets/image/disconiction.jpg",
      }
    }
    style={
      {
        "height": "50%",
        "width": "80%",
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "textAlign": "left",
        },
        {
          "fontFamily": "Roboto-Bold",
          "fontSize": 20,
          "marginTop": 8,
        },
      ]
    }
  >
    app:titles.connectionError
  </Text>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "textAlign": "left",
        },
        {
          "fontFamily": "Roboto-Medium",
          "fontSize": 16,
          "marginTop": 8,
        },
      ]
    }
  >
    app:titles.oops
     
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#007864",
        "height": 82,
        "justifyContent": "center",
        "marginTop": 33,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Bold",
          "fontSize": 20,
        }
      }
    >
      app:buttons.retry
    </Text>
  </View>
</View>
`;
